Packs/HashiCorp-Vault/Integrations/HashiCorpVault/HashiCorpVault.py

```diff
@@ -723,6 +731,59 @@ def get_ch_secrets(engine_path, concat_username_to_cred_name=False):
     return secrets
 
 
+def get_aws_secrets(engine_path, ttl, concat_username_to_cred_name):
+    secrets = []
+    roles_list_url = engine_path + '/roles?list=true'
+    demisto.debug('roles_list_url: {}'.format(roles_list_url))
+    params = {'list': 'true'}
+    res = send_request(roles_list_url, 'get', params=params)
+    if not res or 'data' not in res:
+        return []
+    for role in res['data'].get('keys', []):
+        integration_context = get_integration_context()
+        now = datetime.now()
+        if f'{role}_ttl' in integration_context:
+            last = datetime.fromtimestamp(integration_context[f'{role}_ttl'])
+            diff = (now - last).seconds
+            if diff <= ttl - AWS_TOKEN_OVERLAP_TIME:
+                continue
+        integration_context[f'{role}_ttl'] = now.timestamp()
+        demisto.setIntegrationContext(integration_context)
```

Comment:
Please use set_integration_context
```suggestion
        set_integration_context(integration_context)
```